### PR TITLE
Test result attachment errors

### PR DESCRIPTION
**Pull request by KyleMacPherson**
https://trello.com/c/6NkU8dEZ/857-test-result-audit-log-missing-some-info

## Description
Various fixes around the audit log for test results

## Checklist:
- [ ] Have you documented your changes in the pull request description?
- [ ] Does the change present any security considerations?
- [ ] Is any gem functionality overloaded? Eg: Devise controller methods being overloaded.
- [ ] Has acceptance criteria been tested by a peer?
- [ ] Has the CHANGELOG been updated? (If change is worth telling users about.)

### General testing (author)
- [ ] Test without JavaScript
- [ ] Test on small screen

### Accessibility testing (author)
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable CSS - does content make sense and appear in a logical order?

### CODE REVIEW

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on review app, looking good as expected.

File: app/models/audit_activity/test/test_result_updated.rb [hunk 1]
```diff
@@ -32,7 +32,9 @@ def new_result
   end
 
   def new_details
-    updates["details"]&.second
+    if updates["details"]&.second
+      updates["details"]&.second.presence || "Removed"
```
Comment (slorek) [inline app/models/audit_activity/test/test_result_updated.rb:36]: I think we will need a migration for existing activity records to nilify the values where they are empty for this to work.

**Review by slorek — CHANGES_REQUESTED** [review]
Looks good, just need to think about migrating existing data I think.

File: app/models/audit_activity/test/test_result_updated.rb (hunk 1 above)
Comment (KyleMacPherson) [reply app/models/audit_activity/test/test_result_updated.rb:36]: Ok after some research we don't actually need this after all:
``` 
"".presence => nil
"x".presence => "x"
```

**Review by slorek — APPROVED** [review]
Migration not needed after all.

**Pull request merged**